### DISCUSSION TITLE
C++ Tutorial tweaks

### DISCUSSION THREAD

**Discussion by ghost** [whippingdot/Language-Tutorials#14] [Add Page]
There's a good amount of the C++ tutorial that could be improved as to not alientate readers right off the bat.

For example, Advanced-Operators.md shows `std::cout` logging `0` and `1` for booleans.

[In my fork](https://github.com/CrimsonCodes0/Language-Tutorials/blob/main/C%2B%2B/4.%20Advanced-Operators.md), I added `std::boolalpha` so the code would show `true` or `false`, which are much more "human-friendly," especially helpful if you're learning the language (which is the audience that tutorials tend to target).

Also, the entire file seems like a misnomer, as the operatiors aren't that "advanced," especially considering how there is a whole plethora of C++ operators that hadn't been covered yet.

Besides that, there's plenty of small errors, such as calling `cout` and `cin` keywords in Setup.md, instead of globals, or multi-level comments instead of multi-line comments.

**Comment by whippingdot** [comment]
Umm can you create a pr with all your changes. I am sorry I made all of these. At first I thought that it outputs `True` or `False` in the console but when I tried the program out it didn't. I like your boolalpha thing - but you have to explain what that does after the code block. Please do so. Also fix the other things like the `cout` and `cin` as I didn't understand what you were saying

**Reply by ghost** [reply]
Do you have any ideas for what you'd add next? I can probably do a quick page or two tommorow.

**Comment by whippingdot** [comment]
Also I meant advanced cause beginners don't really need to learn all these when they first learn about operators.

**Comment by ghost** [comment]
Sure, I'll make the adjustments in a PR in roughly 14-16 hours.

As for `cout` and `cin`, they are merely global variables, not keywords.

Keywords are, for lack of a better word, permeant, that is, they cannot be imported from a file, that fundamentally makes no sense. Keywords are valid anywhere in the language (unless syntatically prohibited), whereas variables are scoped identifiers.

```c++
namespace std {
    std::ostream cout = ...;
    std::istream cin = ...;
}
```
No "magic" whatsoever, they're implemented in plain C++.

**Reply by whippingdot** [reply]
hey i changed this into a discussion cause it seemed more appropriate. For now I am not contributing to this but I might help you with some changes when free. If so I will reference this in a new issue (lul I am just trying to do everything github does in their keynote from last year)

**Reply by ghost** [reply]
I think that you should take a look at this user's usage of issues vs discussions: https://github.com/Jack-Works/proposal-readonly-arraybuffer/pull/2#issuecomment-813803507.

For example, the discussion about adding a CSS tutorial is a very focused topic that can be seen with a goal-orientated focus towards just getting a CSS tutorial onto the repository.

Once the tutorial is created, the discussion is obsolete, that is normally when one would close the relevant issue, which cannot be done with discussions.

**Reply by whippingdot** [reply]
Cool, but we can change this into an issue / reference this in a new issue if we want to implement the feature.

**Reply by ghost** [reply]
Well, this discussion is already done, as the request was already implemented.

**Reply by whippingdot** [reply]
O next time we shall change the discussion into an issue. Next time let us do it right when we decide that we are going to work on the issue.

**Comment by ghost** [comment]
@whippingdot

I saw your comment in Operators.md:
> If anyone find a way to fix this and make it show the decimal please tell me. I thought double or float would work but they don't too

Operations don't care about where the value is being stored, it has **no clue** where it is being stored.

All that matters are the operands, and the operation, nothing more.

What you probably wanted would be this:
```c++
double three = 3.0;
double four = 4.0;

double addition = three + four;
double subtraction = four - three;
double multiplication = three * four;
double division = four / three;
```

all that matters is `double three` and `double four`.

Technicially, no truncuation was ever performed, since the decimals were never even calculated to begin with.

There is one con to using doubles/floats for this section though: you'd have to omit the `%` operator, since it doesn't apply to floats, only integers. (it's a tad early to introduce function calls, but `std::fmod` or `std::remainder` would work)

**Reply by ghost** [reply]
Update: I did it in the PR, but now the section really looks *tiny* :/

**Reply by whippingdot** [reply]
revert it back if you think you should

**Reply by ghost** [reply]
I'm not sure, but either way, the page was small, so maybe it needs some adjustments regardless?

I'll leave it up to whoever reviews the code to make the immediate decision, but a follow-up PR should be done to give it more content.
